go/vt/tabletserver/query_executor.go

```diff
@@ -325,6 +335,69 @@ func (qre *QueryExecutor) execPKIN() (*sqltypes.Result, error) {
 	return qre.fetchMulti(pkRows, limit)
 }
 
+func (qre *QueryExecutor) execNextval() (*sqltypes.Result, error) {
+	t := qre.plan.TableInfo
+	t.Seq.Lock()
+	defer t.Seq.Unlock()
+	if t.NextVal >= t.LastVal {
+		_, err := qre.execAsTransaction(func(conn *TxConnection) (*sqltypes.Result, error) {
+			query := fmt.Sprintf("select next_id, cache, increment from `%s` where id = 0 for update", qre.plan.TableName)
+			conn.RecordQuery(query)
+			qr, err := qre.execSQL(conn, query, false)
+			if err != nil {
+				return nil, err
+			}
+			if len(qr.Rows) != 1 {
+				return nil, fmt.Errorf("unexpected rows from reading sequence %s: %d", qre.plan.TableName, len(qr.Rows))
```

Comment:
Maybe make this error message a bit more explicit, like add a few words:
(are you targeting the right keyspace/shard for this sequence?)